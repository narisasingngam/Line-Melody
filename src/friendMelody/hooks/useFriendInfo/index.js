import { useEffect, useState } from "react";
import axios from "axios";
import { BASE_API } from "../../constants";

export default function useFriendInfo() {
  const [friends, setfriend] = useState({});

  useEffect(() => {
    async function fetchData() {
      await axios.get(BASE_API + "api/friends").then((respones) => {
        setfriend(respones.data);
      });
    }

    fetchData();
  }, [setfriend]);

  return { friends };
}
