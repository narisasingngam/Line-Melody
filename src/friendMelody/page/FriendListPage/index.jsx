import React from "react";
import useFriendInfo from "./../../hooks/useFriendInfo";
import Card from "./../../components/Card";
import CardProfile from './../../components/CardProfile'
import { Container } from "./styled";

const FriendListPage =  () => {
  const { friends } =  useFriendInfo();

  return (
    <Container>
      {friends.items && friends.items.map((friend, index) => (
        <Card>
          <CardProfile name={friend.displayName} image={friend.pictureUrl} userId={friend.userId}/>
        </Card>
      ))}
    </Container>
  );
};

export default FriendListPage;
