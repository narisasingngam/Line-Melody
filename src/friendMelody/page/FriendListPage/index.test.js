import React from "react";
import {render} from '@testing-library/react'
import FriendListPage from ".";
import useFriendInfo from './../../hooks/useFriendInfo'
import 'jest-styled-components'

jest.mock("./../../hooks/useFriendInfo")

describe("FriendListPage", () => {



    beforeEach(() => {
        useFriendInfo.mockReturnValue({
            friends: {
                items: []
            }
        })
    })

  it("renders FriendListPage correctly", () => {
    const {container} = render(<FriendListPage/>)

    expect(useFriendInfo).toBeCalled()

    expect(container.firstChild).toMatchSnapshot()
  });
});