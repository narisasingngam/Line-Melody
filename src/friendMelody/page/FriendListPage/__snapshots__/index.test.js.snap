// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendListPage renders FriendListPage correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 50px;
}

<div
  class="c0"
/>
`;
